octobot_trading/producers/funding_updater.py

```diff
@@ -0,0 +1,79 @@
+# pylint: disable=E0611
+#  Drakkar-Software OctoBot-Trading
+#  Copyright (c) Drakkar-Software, All rights reserved.
+#
+#  This library is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU Lesser General Public
+#  License as published by the Free Software Foundation; either
+#  version 3.0 of the License, or (at your option) any later version.
+#
+#  This library is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+#  Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+import asyncio
+import time
+
+from ccxt.base.errors import NotSupported
+from octobot_commons.constants import HOURS_TO_SECONDS
+
+from octobot_trading.constants import FUNDING_CHANNEL
+from octobot_trading.channels.funding import FundingProducer
+from octobot_trading.enums import ExchangeConstantsFundingColumns
+
+
+class FundingUpdater(FundingProducer):
+    CHANNEL_NAME = FUNDING_CHANNEL
+
+    FUNDING_REFRESH_TIME = 2 * HOURS_TO_SECONDS
+    FUNDING_REFRESH_TIME_MIN = 0.2 * HOURS_TO_SECONDS
+    FUNDING_REFRESH_TIME_MAX = 8 * HOURS_TO_SECONDS
+
+    def __init__(self, channel):
+        super().__init__(channel)
+
+    async def start(self):
+        if not self._should_run():
+            return
+
+        while not self.should_stop and not self.channel.is_paused:
+            next_funding_time = None
+            sleep_time = self.FUNDING_REFRESH_TIME
+            try:
+                for pair in self.channel.exchange_manager.exchange_config.traded_symbol_pairs:
+                    funding: dict = await self.channel.exchange_manager.exchange.get_funding_rate(pair)
+
+                    if funding:
+                        next_funding_time = funding[ExchangeConstantsFundingColumns.NEXT_FUNDING_TIME.value]
+                        await self.push(symbol=pair,
+                                        funding_rate=funding[ExchangeConstantsFundingColumns.FUNDING_RATE.value],
+                                        next_funding_time=next_funding_time,
+                                        timestamp=funding[ExchangeConstantsFundingColumns.TIMESTAMP.value])
+            except NotSupported:
+                self.logger.warning(f"{self.channel.exchange_manager.exchange_name} is not supporting updates")
+                await self.pause()
+            except Exception as e:
+                sleep_time = self.FUNDING_REFRESH_TIME_MIN
+                self.logger.exception(e, True, f"Fail to update funding rate : {e}")
+            finally:
+                if next_funding_time:
+                    should_sleep_time = next_funding_time - time.time()
```

Comment:
We should add a try/except in the updater loop otherwise if there is an error with the 1st founding update, the other ones will never be updated before the next update found (but they still occured on exchange)